### ISSUE TITLE
Condition IsSatisfied Calls

### ISSUE THREAD

**Issue by Noggog** [Mutagen-Modding/Mutagen#231]
Would probably require #98 where each condition type is its own subclass /w strongly typed fields

Then `IsSatisfied` could be added to each subclass, with the appropriate rhs fields as parameters.

A `HasPerk` function could offer a `IsSatisfied` that took in an `INpcGetter` to return a bool, etc.

-----

if it made progress, it might be slow and incremental as people were interested in certain functions

Maybe a side "realm" could be slowly developed.  Eg, conditions would be left in their raw exposure as they are now, but if a user was interested, they could do:
```
object? fancierCondition = magicEffect.Conditions.ToTypedConditionSubclass();
if (fancierCondition == null) throw?
switch (fancierCondition)
{
    case HasPerkCondition hasPerk:
        return hasPerk.IsSatisfied(someNpc);
    default:
        .../
}
```

But then the current API wouldnt break for people using it, and the subclasses could be defined one by one as people were interested in them
with maybe a translation back:
`magicEffect.Condition = hasPerk.ToRawCondition();`

**Comment by Elscrux** [comment]
That's a great idea!
We could also add an extension class that evaluates a while list of conditions.
It would need a lot of thought and work, as conditions can be applied to  so many things. And it's not always clear in a static observation of the game files as dynamic circumstances are needed to evaluate some things. 

There are multiple "Run on" types for each condition, which determines on what record the condition is run on.
### Subject
The condition run the condition on the record that holds the record. This already has some big problems. Dialogue uses conditions to restrict who is able to say what dialogue. So you have something like Subject.GetIsId MaiqTheLiar. But how this condition evaluates depends on the npc you currently talk to. When you talk to Maiq, this evaluates to true, but anybody else false. So from a static inspection of the code this this can be evaluated as it depends on what you do in game. At least not without more context. 
Another condition like this is Subject.GetAngle X . This fully depends on where the actor is currently going in game.
 
### Target
Here it gets even more problematic as the "target" of a condition is also a very dynamic construct that isn't always clearly defined, but in core the problem from above is applied here too. There is also Combat Target as another problematic type that ties into the same issues.

### Quest Alias
This run on type is also very dynamic, as whatever placed record is filling the alias can be fully dynamic and depends on what you do in game. So for example there is a quest that starts when you kill a unique npc. It fills the alias with a relative of that npc who will duel you next time you see them. We can't know from the outside what this is.
So much for this. It means we can't realistically evaluate all things statically.


Of course we can try to mitigate that by giving you a context, but there are enough cases like GetAngle, that won't work without the actual runtime context.
So in conclusion we could add this satisfied call for a few, selected things, but it's not realistic that we can build a proper framework for this.
